src/AbstractApplication.php

```diff
@@ -131,13 +158,54 @@ public function get($key, $default = null)
 	 */
 	public function getLogger()
 	{
-		// If a logger hasn't been set, use NullLogger
-		if (! ($this->logger instanceof LoggerInterface))
+		return $this->getContainer()->get('Psr\\Log\\LoggerInterface');
+	}
+
+	/**
+	 * Get the config of the application.
+	 *
+	 * @return  Joomla\Registry\Registry
+	 *
+	 * @since   2.0
+	 */
+	public function getConfig()
+	{
+		return $this->getContainer()->get('Joomla\\Registry\\Registry');
```

Comment:
This doesn't feel right. Registry can be used in a lot more places than this and from here on the Container will always associate it with the application configuration. I understand why you've had to do it this way. But imho this is not a solution that works in the grander application stack.

Just as an example if you call `$container->buildObject('\PATH\TO\A\MODEL');`, the model constructor is a Registry object that contains the model state - but it's a completely different `Registry` object to that of the application config.